### PR TITLE
feat(mesh): apply known remote tenant deltas instead of just logging

### PR DESCRIPTION
**Pull request by CatherineSue**
### Problem

After #1444 the inbound `td:` subscription routes deltas through `TreeHandle::contains_hash`: known hashes log at `trace!`, unknown hashes trigger a `tree:req:` repair. But the known branch never actually mutates the local tree. With the adapter scheduled to be wired into `server.rs` next, that gap would mean cluster deltas resolve correctly but the receiver's tree never gains the remote tenant — defeating the whole point of mesh sync.

### Solution

Reshape `TreeHandle`: drop `contains_hash` (which only ever answered the routing decision) and replace with one method that does the full thing.

```rust
fn apply_known_remote_insert(
    &self,
    model_id: &str,
    tree_kind: TreeKind,
    node_hash: u64,
    worker_url: &str,
) -> bool;
```

Returns `true` on apply, `false` on unknown. The adapter's existing slice-c repair trigger keys off the `false` return. The matched value (the stored prefix or token sequence) never escapes the trait, which keeps eviction's lifecycle ownership inside `CacheAwarePolicy`.

`CacheAwarePolicy`'s impl walks the model-scoped `hash_index`, fetches the matched prefix (string) or full tokens (token), and calls `string_tree.insert_text` / `token_tree.insert_tokens` under `worker_url`. The string-side matched-prefix comes from the existing `select_worker` populate sites; remote-replay populate (#1364's `apply_insert_to_trees`) stores the full text/tokens — both shapes are re-insertable correctly.

## Changes

- `model_gateway/src/policies/cache_aware.rs`:
  - `TreeHandle::contains_hash` → `TreeHandle::apply_known_remote_insert`.
  - Real impl on `CacheAwarePolicy` doing the model-scoped lookup + tree mutation.
  - Updated the `apply_insert_to_trees` doc to reference the new method name.
- `model_gateway/src/mesh/adapters/tree_sync.rs`:
  - Inbound subscription's known branch now calls `apply_known_remote_insert`.
  - Module doc updated to describe apply (not just classify).
- Tests:
  - `MockTreeHandle` gains an `applied_calls()` log so adapter tests can assert which deltas were applied vs ignored.
  - New `test_apply_known_remote_insert_round_trip` on the policy exercises the real impl end-to-end (known/unknown/wrong-kind/unknown-model).
  - Existing `handle_incoming_batch_consults_tree` rewritten as `handle_incoming_batch_applies_known_and_skips_unknown` with applied-call assertions.

### Not in this PR

Reserved for the rest of the wiring sequence:

**Slice d-2 (next)** — repair roundtrip:
- `tree:req:` subscriber + `iter_repair_pages` walkers (string + token).
- `tree:page:` subscriber + `apply_repair_page`.
- Session lifecycle (re-introduces `RepairProgress` with real readers — page matching, timeout retry).

**Slice d-3** — server wiring:
- `mesh_v2_enabled` flag + adapter construction.
- Cold-start fan-out (`request_repair_for_all_local_models`).
- Connecting `CacheAwarePolicy::sync_insert_*` to `adapter.on_local_insert`.

## Test Plan

- `cargo test -p smg --lib` — 735 tests pass.
- `cargo clippy --workspace --all-targets --all-features -- -D warnings` — clean.
- New unit test on `CacheAwarePolicy::apply_known_remote_insert` covers known/unknown/wrong-kind/unknown-model paths, both `String` and `Token` kinds, and confirms model + kind scope the index correctly.

<details>
<summary>Checklist</summary>

- [x] `cargo +nightly fmt` passes
- [x] `cargo clippy --all-targets --all-features -- -D warnings` passes
- [ ] (Optional) Documentation updated
- [ ] (Optional) Please join us on Slack [#sig-smg](https://slack.lightseek.org) to discuss, review, and merge PRs

</details>

## Summary by CodeRabbit

* **Improvements**
  * More efficient tree synchronization: verified remote updates are applied directly, reducing redundant checks and improving update throughput.
  * Robustness: unknown or mismatched hashes still trigger repair requests and warnings to preserve data consistency.

* **Tests**
  * Updated test coverage for the new apply-path and edge cases (known vs unknown/mismatched entries).

### CODE REVIEW

File: model_gateway/src/policies/cache_aware.rs [hunk 1]
```diff
@@ -704,25 +704,69 @@ pub enum TreeKind {
...
+        };
+        match tree_kind {
+            TreeKind::String => {
+                let Some(path) = model_entry.string_tree.get(&node_hash) else {
+                    return false;
+                };
+                let Some(tree) = self.string_trees.get(model_id) else {
+                    // Hash index entry without a tree is a
+                    // populate-site bug; nothing to apply against.
+                    return false;
```
[the remark this answers is not in the document]
Comment (CatherineSue) [reply model_gateway/src/policies/cache_aware.rs:764]: Fixed in 26dfbb05 — both guard sites now `warn!` with `model_id` and `node_hash` so the populate-site invariant violation is visible instead of masked as a spurious repair.

File: model_gateway/src/policies/cache_aware.rs [hunk 2]
```diff
@@ -1382,6 +1426,79 @@ mod tests {
...
+            0xDEAD_BEEF,
+            "http://w2",
+        ));
+        assert!(!policy.apply_known_remote_insert(
+            "unknown_model",
+            TreeKind::String,
+            text_hash,
+            "http://w2",
+        ));
+    }
```
[the remark this answers is not in the document]
Comment (CatherineSue) [reply model_gateway/src/policies/cache_aware.rs:1515]: Fixed in f8fe5d51 — added `test_apply_known_remote_insert_from_request_hot_path` that seeds the index via `select_worker` for both kinds (matched-prefix shape) and asserts apply succeeds. The original full-path test stays as the apply-replay-shape coverage.
